### PR TITLE
Form dallas

### PR DESCRIPTION
**Pull request by dsloan92**
Created the ability to edit a email template to change the subject, body, and default attachments. There is now a JSON file that controls everything in the email template. Mail file was refactored to use the JSON file as needed. Also, when a file is removed as a default attachment, the file is deleted from the upload directory.

### CODE REVIEW

File: model/validate.php [hunk 1]
```diff
@@ -220,11 +217,14 @@ function validateImageUpload($file, $newName)
...
-                $isValid = false;
-            } else {
+            //since we have the used the tutor's id and name as image file name to store in our db
+            // we know file name is going to be unique so no need check for duplicates
+//            if (file_exists($dirName . $newName)) {
+//                echo $dirName.$newName;
+//                $f3->set("errors['duplicatedImage']", "Sorry! This image is already exist choose another one");
+//                $isValid = false;
+//            }
+        else {
```
Comment (kellerflint) [inline model/validate.php]: Something looks very odd about the indentation levels in this function.

File: scripts/tutorsScript.js [hunk 2]
```diff
@@ -159,10 +159,15 @@ $(".import").on("click", function () {
     if (result) {
         let user_id = $(this).data("userid");
 
-        // ajax deletion
+        // ajax importing
         $.post("../tutorsAjax", {
             user_id: user_id
         });
+        //hide Import button after displaying success message
```
Comment (kellerflint) [inline scripts/tutorsScript.js:166]: This is an excellent idea

File: styles/tutorsStyle.css [hunk 3]
```diff
@@ -26,3 +29,6 @@ th {
     font-size: .7em;
 }
 
+.btn-file input[type=file] {
```
Comment (kellerflint) [inline styles/tutorsStyle.css]: I don't know how alphabetizing style rules works with 2 selectors. Maybe these should go in order but im not sure

**Review by kellerflint — COMMENTED** [review]
Overall this looks great. I appreciate having the email save functionality done via JSON rather than database modifications, it's smart and a great way to avoid more refactoring. I have two comments about indenting and CSS formatting at specific lines but this is very clear overall.

I'm also a little confused about why we have both an upload and uploads folder. Can they be the same thing? If they can't, can they be called something that makes it more clear what the upload folder is doing?

There is an image file being pushed in this branch that should be excluded.

File: model/validate.php (hunk 1 above)
Comment (dsloan92) [reply model/validate.php]: Should be fixed

File: styles/tutorsStyle.css (hunk 3 above)
Comment (dsloan92) [reply styles/tutorsStyle.css]: This has been re-ordered

**Review by dsloan92 — COMMENTED** [review]
Changes should be made. Thanks Keller for the review!

**Review by kellerflint — COMMENTED** [review]
Remove upload from git ignore if it's no longer being used. All good otherwise

**Review by kellerflint — APPROVED** [review]
Much less confusing 👍

**Pull request merged**